src/core/observables/ParticleTraits.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef OBSERVABLES_PARTICLE_TRAITS
+#define OBSERVABLES_PARTICLE_TRAITS
+
+#include "Particle.hpp"
+#include "config.hpp"
+
+namespace GenObs {
+template <> struct traits<const ::Particle *> {
```

Comment:
I don't think this should be cv qualified... In general I  find that specialization for a pointer type awkward. If you happen to not have a pointer, than ... you add an additional specialization?